src/mlpack/tests/string_encoding_test.cpp

```diff
@@ -0,0 +1,1111 @@
+  arma::mat target = {
+    { 1.2876820724517808, 1.6931471805599454, 2.1802352704293200, 0, 0 },
+    { 0, 1.6931471805599454, 2.1802352704293200, 2.1802352704293200,
+      1.6931471805599454 },
+    { 1.2876820724517808, 1, 0, 1.2876820724517808, 0 }
+  };
+  CheckMatrices(output, target, 1e-12);
+}
+
+/**
+ * Test TFIDF encoding for characters, using sublinear
+ * as weighting scheme and smoothidf as false.
+ */
+BOOST_AUTO_TEST_CASE(SublinearTfIdfEncodingIndividualCharactersTest)
+{
+  vector<string> input = {
+    "GACCA",
+    "ABCABCD",
+    "GAB"
+  };
+
+  arma::mat output;
+  TfIdfEncoding<CharExtract::TokenType>
+      encoder(TfIdfEncodingPolicy::TfTypes::SUBLINEAR_TF, false);
+
+  // Passing a empty string to encode charactersrawcountsmoothidftrue
+  encoder.Encode(input, output, CharExtract());
+  arma::mat target = {
+    { 1.4054651081081644, 1.6931471805599454, 2.3796592851687173, 0, 0 },
+    { 0, 1.6931471805599454, 2.3796592851687173, 2.3796592851687173,
+      2.0986122886681100 },
+    { 1.4054651081081644, 1, 0, 1.4054651081081644, 0 }
+  };
+  CheckMatrices(output, target, 1e-12);
+}
+
+/**
+ * Test TFIDF encoding for characters, using term
+ * Frequency as weighting scheme and smoothidf as true.
+ */
+BOOST_AUTO_TEST_CASE(TermFrequencySmoothIdfEncodingIndividualCharactersTest)
+{
+  vector<string> input = {
+    "GACCA",
+    "ABCABCD",
+    "GAB"
+  };
+
+  arma::mat output;
+  TfIdfEncoding<CharExtract::TokenType>
+      encoder(TfIdfEncodingPolicy::TfTypes::TERM_FREQUENCY, true);
+
+  // Passing a empty string to encode charactersrawcountsmoothidftrue
+  encoder.Encode(input, output, CharExtract());
+  arma::mat target = {
+    { 0.2575364144903562, 0.4, 0.5150728289807124, 0, 0 },
+    { 0, 0.2857142857142857, 0.3679091635576516, 0.3679091635576516,
+      0.2418781686514208 },
+    { 0.4292273574839269, 0.3333333333333333, 0, 0.4292273574839269, 0 }
+  };
+  CheckMatrices(output, target, 1e-12);
+}
+
+/**
+ * Test TFIDF encoding for characters, using Term
+ * Frequency as weighting scheme and smoothidf as false.
+ */
+BOOST_AUTO_TEST_CASE(TermFrequencyTfIdfEncodingIndividualCharactersTest)
+{
+  vector<string> input = {
+    "GACCA",
+    "ABCABCD",
+    "GAB"
+  };
+
+  arma::mat output;
+  TfIdfEncoding<CharExtract::TokenType>
+      encoder(TfIdfEncodingPolicy::TfTypes::TERM_FREQUENCY, false);
+
+  // Passing a empty string to encode charactersrawcountsmoothidftrue
+  encoder.Encode(input, output, CharExtract());
+  arma::mat target = {
+    { 0.2810930216216329, 0.4, 0.5621860432432658, 0, 0 },
+    { 0, 0.2857142857142857, 0.4015614594594755, 0.4015614594594755,
+      0.2998017555240157 },
+    { 0.4684883693693881, 0.3333333333333333, 0, 0.4684883693693881, 0 }
+  };
+  CheckMatrices(output, target, 1e-12);
+}
+
+/**
+ * Serialization test for the TF-IDF encoding algorithm with
+ * the CharExtract tokenizer.
+ */
+BOOST_AUTO_TEST_CASE(CharExtractTfIdfEncodingSerialization)
```

Comment:
I think we don't need it since we have the `SplitByAnyOf` test.

Reply:
Yes, but you forgot to serialise the size , data member for overload dictionary_encoding class for int type and hence it was not working smoothly , I have corrected that and add the line , so may be this could come in handy in future, still if you want I will remove it.

Reply:
Yes, I see. Actually, I wrote the test (I mean `CharExtractDictionaryEncodingSerialization`), but I didn't check the size in `CheckDictionaries<int>()`.

I think the best way is to fix `CheckDictionarie4s<int>()`:
```c++
template<>
void CheckDictionaries(const StringEncodingDictionary<int>& expected,
                       const StringEncodingDictionary<int>& obtained)
{
  // MapType is equal to std::arry<size_t, 256>.
  using MapType = typename StringEncodingDictionary<int>::MapType;

  BOOST_REQUIRE_EQUAL(expected.Size(), obtained.Size());

  const MapType& expectedMapping = expected.Mapping();
  const MapType& mapping = obtained.Mapping();

  // And so on...
```